src/wallet/api/wallet.cpp

```diff
@@ -720,9 +720,15 @@ bool WalletImpl::connectToDaemon()
     return result;
 }
 
-bool WalletImpl::connected() const
-{
-    return m_wallet->check_connection();
+Wallet::ConnectionStatus WalletImpl::connected() const
+{
+    bool same_version = false;
+    bool is_connected = m_wallet->check_connection(&same_version);
+    if (!is_connected)
+        return Wallet::ConnectionStatus_Disconnected;
+    if (!same_version)
```

Comment:
Just a question - if the check on line 726 fails because it's not the same version, the function will surely return on line 728 and never get as far as returning ConnectionStatus_WrongVersion?